python/aie/compiler/aiecc/main.py

```diff
@@ -15,12 +15,14 @@
 import platform
 import sys
 import time
-from subprocess import PIPE, run, call
-import tempfile
+import subprocess
 import shutil
-import timeit
 import asyncio
 
+from aie.mlir.passmanager import PassManager
+from aie.mlir.ir import Module, Context, Location
+from aie.dialects import aie as aiedialect
```

Comment:
My main concern here is in how much of a python environment dependency this implies...  For instance, this will make cross compiling much harder, since we now have a python binding dependency.  Also, there are some parts that will still need to be out of process, when we outline each core, so not everything will convert this way?

Reply:
although I tend to agree with this sentiment (if you can pass strings you probably should) cross-compiling the bindings (assuming you have the cross-compiled distro) isn't a [huge lift](https://github.com/makslevental/mlir-wheels/blob/main/.github/workflows/wheels.yml#L176); I plan to add that to bindings wheel matrix shortly.